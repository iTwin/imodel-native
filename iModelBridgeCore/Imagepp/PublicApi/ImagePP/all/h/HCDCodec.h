//:>--------------------------------------------------------------------------------------+
//:>
//:>     $Source: PublicApi/ImagePP/all/h/HCDCodec.h $
//:>
//:>  $Copyright: (c) 2015 Bentley Systems, Incorporated. All rights reserved. $
//:>
//:>+--------------------------------------------------------------------------------------
//-----------------------------------------------------------------------------
// Class : HCDCodec
//-----------------------------------------------------------------------------
// General class for codecs.
//-----------------------------------------------------------------------------

#pragma once

#include <Imagepp/all/h/HFCExclusiveKey.h>

BEGIN_IMAGEPP_NAMESPACE

class HCDPacketRLE;

// Binary optimization: Speed up read/write when codec support direct decompression/compression to/from RLE format.
class HNOVTABLEINIT HCDCodecRLEInterface
    {
public:
    virtual void     DecompressSubsetToRLE(const void* pi_pInData, size_t pi_InDataSize, HFCPtr<HCDPacketRLE>& pio_rpRLEPacket) = 0;
    virtual size_t   CompressSubsetFromRLE(HFCPtr<HCDPacketRLE> const& pi_rpPacketRLE, void* po_pOutBuffer, size_t po_OutBufferSize) = 0;
    };

class HNOVTABLEINIT HCDCodec : public HFCShareableObject<HCDCodec>,
    public HFCExclusiveKey     // to have access to claim/release instead of including m_Key and adding the methods
    {
    HDECLARE_BASECLASS_ID(HCDCodecId_Base)

public:

    // codec states enumeration
    enum State
        {
        STATE_NONE,
        STATE_COMPRESS,
        STATE_DECOMPRESS
        };

    // destructor
    virtual         ~HCDCodec();

    // compression
    virtual size_t          CompressSubset(const void* pi_pInData,
                                           size_t pi_InDataSize,
                                           void* po_pOutBuffer,
                                           size_t pi_OutBufferSize) = 0;

    virtual size_t          DecompressSubset(const void* pi_pInData,
                                             size_t pi_InDataSize,
                                             void* po_pOutBuffer,
                                             size_t pi_OutBufferSize) = 0;

    State                   GetCurrentState() const;
    virtual void            Reset();

    virtual HCDCodec*       Clone() const = 0;

    virtual size_t          GetDataSize() const = 0;

    // subset
    virtual size_t          GetMinimumSubsetSize() const = 0;
    virtual size_t          GetSubsetMaxCompressedSize() const = 0;
    virtual void            SetSubsetSize(size_t pi_Size) = 0;

    // others
    virtual double          GetEstimatedCompressionRatio() const;
    virtual bool            HasRandomAccess() const;
    virtual bool            IsCodecImage() const;
    virtual bool            IsCodecVector() const;
    virtual const WString&  GetName() const;

    virtual HCDCodecRLEInterface* 
                            GetRLEInterface();

protected:

    // constructors
    HCDCodec(const WString& pi_rCodecName);
    HCDCodec(const HCDCodec& pi_rObj);


    // compression
    void                    SetCurrentState(State pi_State);

    WString         m_CodecName;

private:

    State          m_CurrentState;
    };

END_IMAGEPP_NAMESPACE