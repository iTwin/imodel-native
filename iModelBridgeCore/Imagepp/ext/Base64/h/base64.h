#ifndef _base64_h_
#define _base64_h_

#include <iostream>

namespace b64
{
        bool encode(std::istream& input, std::ostream& output);
        bool decode(std::istream& input, std::ostream& output);
};

#endif
