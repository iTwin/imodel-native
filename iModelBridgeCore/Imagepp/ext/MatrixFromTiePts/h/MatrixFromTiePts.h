/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See COPYRIGHT.md in the repository root for full copyright notice.
*--------------------------------------------------------------------------------------------*/

#pragma once


#define TRANSFO_NONE     -1 /* No transformation model */
#define TRANSFO_TRAN_2D   0 /* Translation en x et y */
#define TRANSFO_SIMI_2D   1 /* Translation x,y + Rotation + Echelle */
#define TRANSFO_POLY_2D_1 2 /* Polynomiale deg 1 */
#define TRANSFO_AFFI_2D   3 /* Tran. x,y + Rotation + Echelle + Axes */
#define TRANSFO_PROJ_2D   4 /* Projection d'un plan sur un autre */
#define TRANSFO_POLY_2D_2 5 /* Polynomiale deg 2 */
#define TRANSFO_POLY_2D_3 6 /* Polynomiale deg 3 */
#define TRANSFO_POLY_2D_4 7 /* Polynomiale deg 4 */ /* A tester ??? */
#define TRANSFO_POLY_2D_5 8 /* Polynomiale deg 5 */ /* A tester ??? */
#define TRANSFO_MAT_2D    9 /* Matrice de transformation */
#define TRANSFO_HELM_2D   10
#define TRANSFO_TPS_2D    11
#define TRANSFO_MAX       12

#define TRF_ERROR_MAX_ITERATIONS    -1
#define TRF_ERROR_NO_MEMORY         -2
#define TRF_ERROR_BAD_NUMBER_POINTS -3
#define TRF_ERROR_MATRIX            -4


long GetTransfoMatrixFromScaleAndTiePts (double          po_pMatrix[4][4],
                                         unsigned short  pi_NbVal_GeoTiePoint,
                                         double*         pi_pVal_GeoTiePoint,
                                         unsigned short  pi_NbVal_GeoPixelScale,
                                         double*         pi_pVal_GeoPixelScale);

long GetProjectiveTransfoMatrixFromScaleAndTiePts(double         po_pMatrix[4][4],
                                                  unsigned short pi_NbVal_GeoTiePoint,
                                                  double*        pi_pVal_GeoTiePoint);

long GetAffineTransfoMatrixFromScaleAndTiePts(double         po_pMatrix[4][4],
                                              unsigned short pi_NbVal_GeoTiePoint,
                                              double*        pi_pVal_GeoTiePoint);

long GetTransfoMatrixFromTiePts(double           po_pMatrix[4][4],
                                unsigned short   pi_NbVal_GeoTiePoint,
                                const double*    pi_pVal_GeoTiePoint,
                                long             pi_ModelType);

