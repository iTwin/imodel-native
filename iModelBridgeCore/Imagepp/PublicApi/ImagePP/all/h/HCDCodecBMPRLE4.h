//:>--------------------------------------------------------------------------------------+
//:>
//:>
//:>  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
//:>
//:>+--------------------------------------------------------------------------------------
// Class : HCDCodecBMPRLE8
//-----------------------------------------------------------------------------
// RLE8 codec lass.
//-----------------------------------------------------------------------------
#pragma once

#include "HCDCodecImage.h"

BEGIN_IMAGEPP_NAMESPACE

class HCDCodecBMPRLE4 : public HCDCodecImage
    {
    HDECLARE_CLASS_ID(HCDCodecId_BMPRLE4, HCDCodecImage)

public:

    IMAGEPP_EXPORT                 HCDCodecBMPRLE4();

    HCDCodecBMPRLE4(size_t pi_Width,
                    size_t pi_Height);

    HCDCodecBMPRLE4(const HCDCodecBMPRLE4& pi_rObj);

    ~HCDCodecBMPRLE4();

    size_t          CompressSubset(const void* pi_pInData,
                                   size_t     pi_InDataSize,
                                   void*      po_pOutBuffer,
                                   size_t     pi_OutBufferSize) override;

    size_t          DecompressSubset(const void* pi_pInData,
                                     size_t       pi_InDataSize,
                                     void*        po_pOutBuffer,
                                     size_t       pi_OutBufferSize) override;

    bool           HasLineAccess() const override;

    virtual HCDCodec* Clone() const override;

    bool           IsBitsPerPixelSupported(size_t pi_Bits) const override;

    size_t          GetSubsetMaxCompressedSize() const override;

protected:

private:
    };

END_IMAGEPP_NAMESPACE
