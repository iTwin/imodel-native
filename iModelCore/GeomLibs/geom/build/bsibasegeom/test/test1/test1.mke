#---------------------------------------------------------------------------------------------
#  Copyright (c) Bentley Systems, Incorporated. All rights reserved.
#  See COPYRIGHT.md in the repository root for full copyright notice.
#---------------------------------------------------------------------------------------------
TESTAPP_NAME=test1

%include    $(_MakeFilePath)/../testapp.mki

always:
	cl1.cmd