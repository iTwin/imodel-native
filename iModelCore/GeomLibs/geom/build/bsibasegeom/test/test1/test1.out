 geomlibs test environment -- command line arguments:
 Tail of argv[0] = bsigeomtest1.exe
<OK name="_isTranslate"><int>0</int></OK>
<OK name="bsiPolygon_closestPoint (OUT UP)"><int>1</int></OK>
<OK name="midEdge"><DPoint3d>0,0.5,0</DPoint3d></OK>
<OK name="bsiPolygon_closestPoint (OUTCORNER UP)"><int>1</int></OK>
<OK name="corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="bsiPolygon_closestPoint (IN UP)"><int>2</int></OK>
<OK name="interior"><DPoint3d>0.5,0.5,0</DPoint3d></OK>
<OK name="bsiPolygon_closestPoint (IN UP)"><int>2</int></OK>
<OK name="interior"><DPoint3d>0.5,0.5,0</DPoint3d></OK>
<OK name="chord.EllipseTangent"><double>-8.32667e-017</double></OK>
<OK name="chord.RayTangent"><double>2.22045e-016</double></OK>
<OK name="chord.EllipseTangent"><double>5.55112e-017</double></OK>
<OK name="chord.RayTangent"><double>2.77556e-016</double></OK>
<OK name="bsiTransform_isMirrorAboutPlane (I)"><int>0</int></OK>
<OK name="bsiTransform_isMirrorAboutPlane (origin, normal)"><int>1</int></OK>
<OK name="bsiTransform_isMirrorAboutPlane (origin, normal)"><int>1</int></OK>
<OK name="parsed origin"><DPoint3d>1.73684,1.15789,2.89474</DPoint3d></OK>
<OK name="parsed normal"<DVec3d>0.486664,0.324443,0.811107</DVec3d>/>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="get standard view"><bool::true/></OK>
<OK name="rigid"><bool::true/></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>0</int></OK>
<OK name="rotation angle"><double>0.174533</double></OK>
<OK name="View index"><int>1</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>1</int></OK>
<OK name="rotation angle"><double>0.191986</double></OK>
<OK name="View index"><int>1</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.383972</double></OK>
<OK name="View index"><int>1</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>1</int></OK>
<OK name="rotation angle"><double>0.226893</double></OK>
<OK name="View index"><int>5</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.244346</double></OK>
<OK name="View index"><int>5</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>0</int></OK>
<OK name="rotation angle"><double>0.261799</double></OK>
<OK name="View index"><int>4</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.279253</double></OK>
<OK name="View index"><int>4</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>0</int></OK>
<OK name="rotation angle"><double>0.296706</double></OK>
<OK name="View index"><int>3</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.314159</double></OK>
<OK name="View index"><int>3</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>0</int></OK>
<OK name="rotation angle"><double>0.331613</double></OK>
<OK name="View index"><int>6</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.349066</double></OK>
<OK name="View index"><int>6</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>1</int></OK>
<OK name="rotation angle"><double>0.366519</double></OK>
<OK name="View index"><int>2</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.383972</double></OK>
<OK name="View index"><int>2</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.383972</double></OK>
<OK name="View index"><int>7</int></OK>
<OK name="isRotationFromStandardView"><int>1</int></OK>
<OK name="axis index"><int>2</int></OK>
<OK name="rotation angle"><double>0.401426</double></OK>
<OK name="View index"><int>8</int></OK>
<OK name="Triangle Closest Point Above Interior"><DPoint3d>0.442,0.01,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Interior"><DPoint3d>0.442,0.01,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Edge"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Edge"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Exterior"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Exterior"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Corner"><DPoint3d>0,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Corner"><DPoint3d>0,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Interior"><DPoint3d>0.442,0.01,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Interior"><DPoint3d>0.442,0.01,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Edge"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Edge"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Exterior"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Exterior"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Corner"><DPoint3d>0,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Corner"><DPoint3d>0,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Interior"><DPoint3d>0.442,0.01,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Interior"><DPoint3d>0.442,0.01,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Edge"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Edge"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Exterior"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Exterior"><DPoint3d>0.442,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Corner"><DPoint3d>0,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Corner"><DPoint3d>0,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Interior"><DPoint3d>0.548,0.432,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Interior"><DPoint3d>0.548,0.432,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Edge"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Edge"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Exterior"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Exterior"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Corner"><DPoint3d>1,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Corner"><DPoint3d>1,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Interior"><DPoint3d>0.548,0.432,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Interior"><DPoint3d>0.548,0.432,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Edge"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Edge"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Exterior"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Exterior"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Corner"><DPoint3d>1,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Corner"><DPoint3d>1,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Interior"><DPoint3d>0.548,0.432,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Interior"><DPoint3d>0.548,0.432,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Edge"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Edge"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Exterior"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Exterior"><DPoint3d>0.558,0.442,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Corner"><DPoint3d>1,0,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Corner"><DPoint3d>1,0,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Interior"><DPoint3d>0.01,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Interior"><DPoint3d>0.01,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Edge"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Edge"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Exterior"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Exterior"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Interior"><DPoint3d>0.01,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Interior"><DPoint3d>0.01,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Edge"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Edge"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Exterior"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Exterior"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point ON Corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Interior"><DPoint3d>0.01,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Interior"><DPoint3d>0.01,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Edge"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Edge"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Exterior"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Exterior"><DPoint3d>0,0.558,0</DPoint3d></OK>
<OK name="Triangle Closest Point Below Corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Corner"><DPoint3d>0,1,0</DPoint3d></OK>
<OK name="Triangle Closest Point Above Interior"><DPoint3d>-0.908295,-0.20753,4.06916</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Interior"><DPoint3d>-0.908295,-0.20753,4.06916</DPoint3d></OK>
<OK name="Triangle Closest Point Above Edge"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Edge"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Triangle Closest Point Above Exterior"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Exterior"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Triangle Closest Point Above Corner"><DPoint3d>1,2,3</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Corner"><DPoint3d>1,2,3</DPoint3d></OK>
<OK name="Triangle Closest Point ON Interior"><DPoint3d>-0.908295,-0.20753,4.06916</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Interior"><DPoint3d>-0.908295,-0.20753,4.06916</DPoint3d></OK>
<OK name="Triangle Closest Point ON Edge"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Edge"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Triangle Closest Point ON Exterior"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Exterior"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Triangle Closest Point ON Corner"><DPoint3d>1,2,3</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Corner"><DPoint3d>1,2,3</DPoint3d></OK>
<OK name="Triangle Closest Point Below Interior"><DPoint3d>-0.908295,-0.20753,4.06916</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Interior"><DPoint3d>-0.908295,-0.20753,4.06916</DPoint3d></OK>
<OK name="Triangle Closest Point Below Edge"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Edge"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Triangle Closest Point Below Exterior"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Exterior"><DPoint3d>-0.8564,-0.2542,4.0608</DPoint3d></OK>
<OK name="Triangle Closest Point Below Corner"><DPoint3d>1,2,3</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Corner"><DPoint3d>1,2,3</DPoint3d></OK>
<OK name="Triangle Closest Point Above Interior"><DPoint3d>-1.55993,1.01278,4.05092</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Interior"><DPoint3d>-1.55993,1.01278,4.05092</DPoint3d></OK>
<OK name="Triangle Closest Point Above Edge"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Edge"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Triangle Closest Point Above Exterior"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Exterior"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Triangle Closest Point Above Corner"><DPoint3d>-3.2,-3.1,5.4</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Corner"><DPoint3d>-3.2,-3.1,5.4</DPoint3d></OK>
<OK name="Triangle Closest Point ON Interior"><DPoint3d>-1.55993,1.01278,4.05092</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Interior"><DPoint3d>-1.55993,1.01278,4.05092</DPoint3d></OK>
<OK name="Triangle Closest Point ON Edge"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Edge"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Triangle Closest Point ON Exterior"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Exterior"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Triangle Closest Point ON Corner"><DPoint3d>-3.2,-3.1,5.4</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Corner"><DPoint3d>-3.2,-3.1,5.4</DPoint3d></OK>
<OK name="Triangle Closest Point Below Interior"><DPoint3d>-1.55993,1.01278,4.05092</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Interior"><DPoint3d>-1.55993,1.01278,4.05092</DPoint3d></OK>
<OK name="Triangle Closest Point Below Edge"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Edge"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Triangle Closest Point Below Exterior"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Exterior"><DPoint3d>-1.653,1.0548,4.074</DPoint3d></OK>
<OK name="Triangle Closest Point Below Corner"><DPoint3d>-3.2,-3.1,5.4</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Corner"><DPoint3d>-3.2,-3.1,5.4</DPoint3d></OK>
<OK name="Triangle Closest Point Above Interior"><DPoint3d>0.568222,4.39475,2.67992</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Interior"><DPoint3d>0.568222,4.39475,2.67992</DPoint3d></OK>
<OK name="Triangle Closest Point Above Edge"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Edge"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Triangle Closest Point Above Exterior"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Exterior"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Triangle Closest Point Above Corner"><DPoint3d>0.3,6.3,2.4</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Above Corner"><DPoint3d>0.3,6.3,2.4</DPoint3d></OK>
<OK name="Triangle Closest Point ON Interior"><DPoint3d>0.568222,4.39475,2.67992</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Interior"><DPoint3d>0.568222,4.39475,2.67992</DPoint3d></OK>
<OK name="Triangle Closest Point ON Edge"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Edge"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Triangle Closest Point ON Exterior"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Exterior"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Triangle Closest Point ON Corner"><DPoint3d>0.3,6.3,2.4</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz ON Corner"><DPoint3d>0.3,6.3,2.4</DPoint3d></OK>
<OK name="Triangle Closest Point Below Interior"><DPoint3d>0.568222,4.39475,2.67992</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Interior"><DPoint3d>0.568222,4.39475,2.67992</DPoint3d></OK>
<OK name="Triangle Closest Point Below Edge"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Edge"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Triangle Closest Point Below Exterior"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Exterior"><DPoint3d>0.6094,4.3994,2.6652</DPoint3d></OK>
<OK name="Triangle Closest Point Below Corner"><DPoint3d>0.3,6.3,2.4</DPoint3d></OK>
<OK name="Tri minDist uvw->xyz Below Corner"><DPoint3d>0.3,6.3,2.4</DPoint3d></OK>
<OK name="unit square"><bool::true/></OK>
<OK name="Centroid"><DPoint3d>0.5,0.5,0</DPoint3d></OK>
<OK name="normal"<DVec3d>0,0,1</DVec3d>/>
<OK name="area"><double>1</double></OK>
<OK name="perimeter"><double>4</double></OK>
<OK name="rotated"><bool::true/></OK>
<OK name="Centroid"><DPoint3d>0.251792,-0.190672,0.643904</DPoint3d></OK>
<OK name="normal"<DVec3d>-0.0308105,0.253588,0.966821</DVec3d>/>
<OK name="area"><double>1</double></OK>
<OK name="perimeter"><double>4</double></OK>
<OK name="slit square"><bool::true/></OK>
<OK name="Centroid"><DPoint3d>0.5,0.5,0</DPoint3d></OK>
<OK name="normal"<DVec3d>0,0,1</DVec3d>/>
<OK name="area"><double>1</double></OK>
<OK name="perimeter"><double>6.82843</double></OK>
<OK name="rotated"><bool::true/></OK>
<OK name="Centroid"><DPoint3d>0.251792,-0.190672,0.643904</DPoint3d></OK>
<OK name="normal"<DVec3d>-0.0308105,0.253588,0.966821</DVec3d>/>
<OK name="area"><double>1</double></OK>
<OK name="perimeter"><double>6.82843</double></OK>
<OK name="l blocks"><bool::true/></OK>
<OK name="Centroid"><DPoint3d>0.833333,0.833333,0</DPoint3d></OK>
<OK name="normal"<DVec3d>0,0,1</DVec3d>/>
<OK name="area"><double>3</double></OK>
<OK name="perimeter"><double>8</double></OK>
<OK name="rotated"><bool::true/></OK>
<OK name="Centroid"><DPoint3d>0.632951,0.0803737,0.584958</DPoint3d></OK>
<OK name="normal"<DVec3d>-0.0308105,0.253588,0.966821</DVec3d>/>
<OK name="area"><double>3</double></OK>
<OK name="perimeter"><double>8</double></OK>
Hello world
Exit with 0 errors, 268 OK
